app/views/comments/_form.html.erb

```diff
@@ -66,38 +66,54 @@
 
   <%= javascript_include_tag "dragdrop" %>
   <script>
-    $('#comment-form').bind('ajax:beforeSend', function(event){      
+    $('#comment-form').bind('ajax:beforeSend', function(event){
       $("#text-input").prop('disabled',true)
-      $("#comment-form .btn-primary").button('loading',true);      
+      $("#comment-form .btn-primary").button('loading',true);
     });
-    <% emoji_names = emoji_names_list %>
-    <% emojis = emoji_names[:emojis] %>
-    <% image_map = emoji_names[:image_map] %>
-    var image_map = <%= raw image_map.to_json %>
-    horsey(document.querySelector('textarea#text-input'), {
-        suggestions: (<%= raw emojis.to_json %>),
-        anchor: ':',
-        render: function (li, suggestion) {
-            var image = image_map[suggestion.text];
-            li.innerHTML = "<b>"+image+"</b><span>"+suggestion.text+"</span>";
-        },
-        limit: 6
+
+    names = <%= raw User.pluck(:username).compact.to_json %>;
```

Comment:
Ah, so here, does atwho allow for remote loading of suggestions and especially ones base on what letters you've already typed? That would do a lot to optimize this for when it runs on the full db which has 300k usernames.

Reply:
So for this, see my note above about the API and 

https://publiclab.org/api/srch/profiles?srchString=jy

I think atwho will probably have a way to fetch remote results!

Thanks and great work!

Reply:
Hi @jywarren awesome to learn how to deal with this kind of transaction for a large user community (I haven't run into that before  now ;) I think it should be possible to fetch remote data (e.g. from atwho docs):

When you set data as URL string, At.js will launch Ajax request to the server to get the JSON data

   $('textarea').atwho({
        at: "@", 
        data: "http://www.atjs.com/users.json",
        limit: 7
    });

so I would need to combine that with the API you mention to get the proper url. If that is not possible, I should at least be able to put the code in application_helper, as with the emoji example, but I think the API approach should be feasible. As a side note, my  less than a year old Mac has decided it's display wants to fail so in the next few days, I have to send it to Apple for repairs. If I fall off the radar for a few days, it is because of this, but I'm hoping it'll hold out for a few more days so I can get some work done before sending it off.